File: apps/clubos-web/src/types/events.ts
export type EventCategory =
  | "player_signing"
  | "player_departure"
  | "match_result_win"
  | "match_result_loss"
  | "trophy_win"
  | "trophy_loss"
  | "transfer_window"
  | "media_event"
  | "injury_news"
  | "commercial_event";

export type ImpactMagnitude = "high" | "medium" | "low";

export interface EventSchema {
  event_id: string;
  event_date: string;
  event_name: string;
  event_category: EventCategory;
  event_description: string;
  expected_impact: string; // Comma-separated
  affected_assets: string[];
  impact_magnitude: ImpactMagnitude;
}

export interface EventCreateSchema {
  event_date: string;
  event_name: string;
  event_category: EventCategory;
  event_description: string;
  expected_impact: string;
  impact_magnitude: ImpactMagnitude;
}

export interface EventListResponse {
  total_count: number;
  items: EventSchema[];
}

File: backend/api/app/schemas/events.py
from typing import Optional
from pydantic import BaseModel


class EventSchema(BaseModel):
    event_id: str
    event_date: str  # ISO 8601 date string
    event_name: str
    event_category: str
    event_description: str
    expected_impact: str  # Comma-separated list of affected assets
    affected_assets: list[str]  # Parsed from expected_impact
    impact_magnitude: str  # 'high', 'medium', 'low'


class EventCreateSchema(BaseModel):
    event_date: str
    event_name: str
    event_category: str
    event_description: str
    expected_impact: str  # Comma-separated asset names
    impact_magnitude: str


class EventListResponse(BaseModel):
    total_count: int
    items: list[EventSchema]
